### ISSUE TITLE
[JS] Implicit Any Type for Map Function

### ISSUE THREAD

**Issue by minhanh-phan** [Azure/autorest.typescript#1999]
The DPG generation for @azure/schema-registry package using [this tsp file](https://github.com/minhanh-phan/azure-rest-api-specs/blob/64c0f42845bc61cea5762df36b9915d3ef13fcf9/specification/schemaregistry/SchemaRegistry/main.tsp) does not specify type for parameters in the map function. We expect the variable `p` below to have the correct type.

The issue is at [this line](https://github.com/Azure/azure-sdk-for-js/blob/5e1bca92c54b43bb97f8ffc52a6f0aaf0d700b4e/sdk/schemaregistry/schema-registry/src/api/operations.ts#L120). In this case, `p` should have the type `VersionOutput`.
```ts
return {
  value: (result.body["value"] ?? []).map((p) => ({
    schemaVersion: p["schemaVersion"],
  })),
  nextLink: result.body["nextLink"],
};
```

The issue is at [this line](https://github.com/Azure/azure-sdk-for-js/blob/5e1bca92c54b43bb97f8ffc52a6f0aaf0d700b4e/sdk/schemaregistry/schema-registry/src/api/operations.ts#L53). In this case, `p` should have the type `SchemaGroupOutput`.
```ts
return {
  value: (result.body["value"] ?? []).map((p) => ({
    groupName: p["groupName"],
  })),
  nextLink: result.body["nextLink"],
};
```

For more reference, [here](https://github.com/Azure/azure-sdk-for-js/blob/5e1bca92c54b43bb97f8ffc52a6f0aaf0d700b4e/sdk/schemaregistry/schema-registry/src/api/operations.ts) is the generated file. 

To reproduce:

- Set the `tsp-location.yaml` with these info
```yaml
directory: specification/schemaregistry/SchemaRegistry
commit: 64c0f42845bc61cea5762df36b9915d3ef13fcf9
repo: Azure/azure-rest-api-specs
```
- Generate

**Comment by MaryGao** [comment]
@minhanh-phan I think this is our inner deserialization code and I was wondering if this would cause any errors, e.g build errors?

**Comment by minhanh-phan** [comment]
@MaryGao Yes, it would cause this build error `error TS7006: Parameter 'p' implicitly has an 'any' type.`

**Comment by MaryGao** [comment]
Could you share the ci logs with me? Because I notice that our smoke testing didn't report this issue. Maybe there is difference for the tsconfig.json file and I need to identify the gaps.
